src/devices_models/device_constructors/branch_constructor.jl

```diff
@@ -63,38 +63,34 @@ function construct_device!(
     ::Type{S},
 ) where {B <: PSY.ACBranch, S <: StandardPTDFModel}
     devices = get_available_components(B, sys)
+
     if !validate_available_devices(B, devices)
         return
     end
-
+    @info B
     add_variables!(optimization_container, StandardPTDFModel(), devices)
 
-    # PTDF
+    # PTDF flow calculations
     ptdf = get_PTDF(optimization_container)
     buses = PSY.get_components(PSY.Bus, sys)
+
     time_steps = model_time_steps(optimization_container)
     constraint_val = JuMPConstraintArray(undef, time_steps)
     network_flow =
         add_cons_container!(optimization_container, :network_flow, ptdf.axes[1], time_steps)
 
-    nodal_balance_expressions = optimization_container.expressions[:nodal_balance_active]
-    for t in time_steps
-        for br in devices
-            flow_variable = get_variable(optimization_container, FLOW_ACTIVE_POWER, B)
-            name = PSY.get_name(br)
-            line_flow =
-                model_has_parameters(optimization_container) ? zero(PGAE) :
-                JuMP.AffExpr(0.0)
-            for b in buses
-                bus_number = PSY.get_number(b)
-                _flow = ptdf[name, bus_number] * nodal_balance_expressions[bus_number, t]
-                JuMP.add_to_expression!(line_flow, _flow)
-            end
-            network_flow[name, t] = JuMP.@constraint(
-                optimization_container.JuMPmodel,
-                flow_variable[name, t] == line_flow
-            )
-        end
+    flow_variables = get_variable(optimization_container, FLOW_ACTIVE_POWER, B)
+    nodal_balance_expressions =
+        get_expression(optimization_container, :nodal_balance_active)
+    jump_model = get_jump_model(optimization_container)
+    for t in time_steps, br in devices
+        br_name = get_name(br)
+        network_flow[br_name, t] = JuMP.@constraint(
```

Comment:
@odow I resolved some of the issues we discussed here. WIthout parameters building the lines takes about 115 seconds together between network, transformers, and lines.

![image](https://user-images.githubusercontent.com/16385323/109118257-f2f3dd80-76f7-11eb-92da-df7c25c765eb.png)

with ParameterJuMP Parameters takes 10 times more time 

![image](https://user-images.githubusercontent.com/16385323/109118684-7ca3ab00-76f8-11eb-9058-df769735b342.png)

In the sum ptdf.axes[2] is an array of integers and the function now doesn't do type inferences at run time.

Reply:
Try doing:
```julia
using MutableArithmetics
const MA = MutableArithmetics
expr = zero(MA.promote_operation(MA.add_mul, eltype(flow_variables), eltype(ptdf), eltype(nodal_balance_expressions))
MA.mutable_operate!(+, expr, flow_variables[br_name, t])
for i in ptdf.axes[2]
    MA.mutable_operate!(MA.sub_mul, expr, ptdf[br_name, i], nodal_balance_expressions[i, t])
end
```
The issue with the code written by macros is that it uses `MA.operate!` which falls back to inefficient operation in case none is implemented.
With `MA.mutable_operate!`, you will either have an efficient code or an error, and the error would make it easy to know what is missing in ParameterJuMP